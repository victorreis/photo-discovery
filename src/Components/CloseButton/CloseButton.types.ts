import { TestProps } from '../../Config/Tests/Test.types';

export interface RequiredCloseButtonProps {
  onClick: () => void;
}

export interface DefaultCloseButtonProps {}

export interface OptionalCloseButtonProps {}

export type CloseButtonProps = RequiredCloseButtonProps &
  DefaultCloseButtonProps &
  OptionalCloseButtonProps &
  TestProps &
  Omit<React.HTMLAttributes<HTMLButtonElement>, 'children' | 'onClick'>;

export type CloseButtonStyleProps = Required<DefaultCloseButtonProps>;
