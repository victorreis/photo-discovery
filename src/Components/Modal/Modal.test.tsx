import {
  fireEvent,
  renderJestDomCreator,
  renderRTRCreator,
  screen,
} from '../../Config/Tests/GlobalSetup.config';
import { closeButtonDefaults } from '../CloseButton/CloseButton';
import { Modal, modalDefaults } from './Modal';
import { RequiredModalProps, ModalProps } from './Modal.types';

describe(`Modal component tests`, () => {
  const text = 'text';
  const onClose = jest.fn();

  const modalBackdropContainerId = `${modalDefaults.testID}_ModalBackdropContainer`;

  const requiredProps: RequiredModalProps = {
    children: text,
  };

  const setup = (props?: ModalProps) => {
    const renderRTR = () =>
      renderRTRCreator<ModalProps>(Modal, {
        ...requiredProps,
        ...props,
      });
    const renderJestDom = () =>
      renderJestDomCreator<ModalProps>(Modal, {
        ...requiredProps,
        ...props,
      });

    return { renderRTR, renderJestDom };
  };

  describe(`behavior tests`, () => {
    it(`should render the component`, () => {
      setup().renderJestDom();
      const testInstance = screen.getByTestId(modalDefaults.testID);

      expect(testInstance).toBeTruthy();
    });

    it(`should render the text`, () => {
      setup().renderJestDom();
      const element = screen.getByText(text);

      expect(element).toBeInTheDocument();
    });

    it(`should call the onCLick callback when the '${modalBackdropContainerId}' is clicked`, () => {
      setup({ ...requiredProps, onClose }).renderJestDom();
      const modalBackdropContainerInstance = screen.getByTestId(
        modalBackdropContainerId
      );

      fireEvent.click(modalBackdropContainerInstance);
      expect(onClose).toHaveBeenCalled();
    });

    it(`should call the onCLick callback when the '${closeButtonDefaults.testID}' is clicked`, () => {
      setup({ ...requiredProps, onClose }).renderJestDom();
      const closeButtonInstance = screen.getByTestId(
        closeButtonDefaults.testID
      );

      fireEvent.click(closeButtonInstance);
      expect(onClose).toHaveBeenCalled();
    });
  });

  describe(`snapshot tests`, () => {
    it(`should render correctly`, () => {
      const generatedJson = setup().renderRTR().toJSON();
      expect(generatedJson).toMatchSnapshot();
    });
  });
});
