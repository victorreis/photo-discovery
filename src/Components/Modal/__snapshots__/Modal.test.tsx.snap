// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modal component tests snapshot tests should render correctly 1`] = `
<div
  className="ModalContainer-photo-discovery__sc-1x2la9l-0 kZrEh"
  data-testid="Modal"
>
  <div
    className="ModalContentContainer-photo-discovery__sc-1x2la9l-2 iljsDR"
    data-testid="Modal_ModalContentContainer"
  >
    <button
      className="CloseButtonContainer-photo-discovery__sc-khdkyf-1 dcJekt"
      data-testid="CloseButton"
      onClick={[Function]}
    >
      <svg
        color="#e8e8ee"
        fill="currentColor"
        height="1.67rem"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "#e8e8ee",
          }
        }
        viewBox="0 0 24 24"
        width="1.67rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"
        />
      </svg>
      <div
        className="TransparentContainer-photo-discovery__sc-khdkyf-0 kjHOyB"
      />
    </button>
    text
  </div>
  <div
    className="ModalBackdropContainer-photo-discovery__sc-1x2la9l-1 iejXpF"
    data-testid="Modal_ModalBackdropContainer"
    onClick={[Function]}
  />
</div>
`;
